### PR TITLE
xclim 0.48 compatibility

### PR DESCRIPTION
**Pull request by RondeauG**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - No
- [x] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [x] (If applicable) Tests have been added.
- [x] This PR does not seem to break the templates.
- [x] CHANGES.rst has been updated (with summary of main changes).
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.
- [x] Maybe... Add a workaround to convert the old xrfreq nomenclature to the new one upon loading an existing catalog or parsing a directory.

### What kind of change does this PR introduce?

* Changes to be compatible with recent versions of `pandas/xarray/xclim`

### Does this PR introduce a breaking change?

- `xscen` now requires `pandas >= 2.2`
- Functions that output a dict with keys as xrfreq (`extract_dataset, compute_indicators`) will now return the new nomenclature.
- The `frequency --> xrfreq/timedelta` will now only result in the new nomenclature for hours (lower case `h`).

### CODE REVIEW

File: environment-dev.yml [hunk 1]
```diff
@@ -21,15 +21,15 @@ dependencies:
...
+  - pandas >=2.2
   - parse
   - pyyaml
   - rechunker
   - scipy
   - shapely >=2.0
   - sparse
   - toolz
-  - xarray <2023.11.0  # FIXME: Remove when pandas 2.2 is released and xclim is fixed.
+  - xarray
```
Comment (Zeitsperre) [inline environment-dev.yml:32]: It might be worthwhile to set the lower bounds for this pin.

File: environment.yml [hunk 2]
```diff
@@ -21,15 +21,15 @@ dependencies:
...
+  - pandas >=2.2
   - parse
   - pyyaml
   - rechunker
   - scipy
   - shapely >=2.0
   - sparse
   - toolz
-  - xarray <2023.11.0  # FIXME: Remove when pandas 2.2 is released and xclim is fixed.
+  - xarray
```
Comment (Zeitsperre) [inline environment.yml:32]: ibid.

File: pyproject.toml [hunk 3]
```diff
@@ -62,7 +62,7 @@ dependencies = [
   "shapely>=2.0",
   "sparse",
   "toolz",
-  "xarray<2023.11.0", # FIXME: Remove when pandas 2.2 is released and xclim is fixed.
+  "xarray",
```
Comment (Zeitsperre) [inline pyproject.toml:65]: ibid.

File: environment-dev.yml (hunk 1 above)
Comment (RondeauG) [reply environment-dev.yml:32]: What would be the right bound, 2023.11.0?

**Comment by juliettelavoie** [comment]
https://github.com/Ouranosinc/xclim/pull/1664 should fix the issue we were having with notebook 3. I have temporarily removed `measure` from the yaml until a new release of xclim.

**Comment by RondeauG** [comment]
Let's wait for https://github.com/Ouranosinc/xclim/pull/1664 before we finish this PR.

**Comment by Zeitsperre** [comment]
The problem with Python3.12 is quite silly. Pushing a fix a few mins.

**Comment by aulemahal** [comment]
I can push something that ensures the new xrfreq is used when we open catalogs (and parse them).

**Comment by aulemahal** [comment]
Last commit added a `ensure_new_xrfreq` function to `utils`. It is used when opening a catalog and in `parse_directory`. This means that legacy frequencies either in the paths or in the catalogs shouldn't be a problem.

Running on MRCC5-disponible, our largest catalog, my `n=1` test finds that the opening time has increased by 6%. I decided that this was small enough not to require further optimization or skipping options.

**Comment by Zeitsperre** [comment]
I'll be releasing xclim v0.48.2 today!

**Comment by Zeitsperre** [comment]
This PR is waiting on the newest patch release from conda-forge. Should be available later today.

**Comment by RondeauG** [comment]
`xclim>=0.48.0` ou `xclim>=0.48.2` ?

File: xscen/CVs/xrfreq_to_timedelta.json [hunk 4]
```diff
@@ -1,7 +1,10 @@
 {
   "is_regex": true,
+  "h": "1h",
   "H": "1H",
+  "3h": "3h",
   "3H": "3H",
+  "6h": "6h",
   "6H": "6H",
```
Comment (aulemahal) [inline xscen/CVs/xrfreq_to_timedelta.json:8]:
```suggestion
  "H": "1h",
  "3h": "3h",
  "3H": "3h",
  "6h": "6h",
  "6H": "6h",
```

File: tox.ini [hunk 5]
```diff
@@ -69,6 +69,7 @@ download = true
 deps =
     coveralls: coveralls
     esmpy: git+https://github.com/esmf-org/esmf.git@v{env:ESMF_VERSION}\#subdirectory=src/addon/esmpy
+    xclim: git+https://github.com/Ouranosinc/xclim.git@main
```
Comment (juliettelavoie) [inline tox.ini:72]: is this still necessary with the new release ?

**Review by aulemahal — APPROVED** [review]
One fix, but the rest seems to be good!

File: xscen/catutils.py [hunk 6]
```diff
@@ -643,6 +648,7 @@ def parse_directory(  # noqa:C901
             df["xrfreq"].apply(CV.xrfreq_to_frequency, default=pd.NA), inplace=True
         )
 
+    # Esu
```
Comment (juliettelavoie) [inline xscen/catutils.py:651]: ?

Comment (aulemahal) [reply xscen/catutils.py:651]: haha woups.
```suggestion
```

File: tox.ini (hunk 5 above)
Comment (Zeitsperre) [reply tox.ini:72]: I figured we could leave it there and have at least one build that checks against the main branch. For xclim, we have a dedicated upstream build; Would it be prudent to implement that here as well? (in another PR)

**Review by juliettelavoie — APPROVED** [review]
lgtm

File: tox.ini (hunk 5 above)
Comment (juliettelavoie) [reply tox.ini:72]: Ah I see. no strong opinions from me for the upstream build.

Comment (Zeitsperre) [reply tox.ini:72]: You say "no strong opinions", I read "no opposition, go for it". You got it. ;)

**Pull request merged**